meshparty/skeleton.py

```diff
@@ -180,17 +185,60 @@ def reroot(self, new_root):
         self._parent_node_array[e1] = e2
         self._reset_derived_objects()
 
+
+    def cut_graph(self, vinds, directed=True, euclidean_weight=True):
+        '''
+        Return a csgraph for the skeleton with each ind in inds cut off from its parent.
+        '''
+        e_keep = ~np.isin(self.edges[:,0], vinds)
+        es_new = self.edges[e_keep]
+        return utils.create_csgraph(self.vertices, es_new,
+                                    euclidean_weight=euclidean_weight,
+                                    directed=directed)
+
+    def downstream_nodes(self, vinds):
+        if np.isscalar(vinds):
+            vinds = [vinds]
+            return_single = True
+        else:
+            return_single = False
+
+        dns = []
+        for vind in vinds:
+            g = self.cut_graph(vinds)
```

Comment:
is cut graph the right idea here? or is it just a directed graph with the edges flipped?

Reply:
the cut graph is fine, but I replaced connected components with Dijkstra on the graph transpose. Much faster for large trees than doing the same thing in python.